portfolio/src/main/webapp/script.js

```diff
@@ -42,3 +42,505 @@ function visitLink(s){
+                        square.col == this.selected_square.col && !selected_piece.moved && 
+                        !square.piece){
+
+                        selected_piece.moved = true;
+                        return true;
+                    }
+
+                    //captures diagonally
+                    else if(square.row == this.selected_square.row + 1 && 
+                        (square.col == this.selected_square.col + 1 || 
+                        square.col == this.selected_square.col - 1) &&
+                        square.piece){
+                        
+                        selected_piece.moved = true;
+                        return true;
+                    }
+                }
+            }
+            else if(selected_piece.type == 'knight'){
+                //Two vertical and one horizontal or two horizontal and one vertical
+                if((Math.abs(square.row - this.selected_square.row) == 2 &&
+                    Math.abs(square.col - this.selected_square.col) == 1) ||
+                    (Math.abs(square.row - this.selected_square.row) == 1 && 
+                    Math.abs(square.col - this.selected_square.col) == 2)){
+                    
+                    return true;
+                }
+            }
+            else if(selected_piece.type == 'bishop'){
+                if(square.row - this.selected_square.row == square.col - this.selected_square.col){
+                    return true;
+                }
+                else if((square.row - this.selected_square.row) + (square.col - this.selected_square.col) == 0){
+                    return true;
+                }
+            }
+            else if(selected_piece.type == 'rook'){
+                //There is no change in horizontal position or there is no vertical change
+                //And 
+                //the row is not the same or the column is not the same
+                if((square.row - this.selected_square.row == 0 || 
+                    square.col - this.selected_square.col == 0) && (square.row != 
+                    this.selected_square.row || square.col != this.selected_square.col)){
+
+                    selected_piece.moved = true;
+                    return true;
+                }
+            }
+            else if(selected_piece.type == 'queen'){
+                //Bishop moving patterns
+                if(square.row - this.selected_square.row == square.col - this.selected_square.col){
+                    return true;
+                }
+                else if((square.row - this.selected_square.row) + 
+                    (square.col - this.selected_square.col) == 0){
+                    return true;
+                }
+                //Rook moving patterns
+                else if((square.row - this.selected_square.row == 0 || 
+                    square.col - this.selected_square.col == 0) && (square.row != 
+                    this.selected_square.row || square.col != this.selected_square.col)){
```

Comment:
Do we need this check that the selected square is not the square since it is handled elsewhere? Same with the rook move above?

Reply:
Also I'm just realizing there's nothing to stop people from jumping over other pieces with the queen, bishop, and rook.

Reply:
Yeah :/